### ISSUE TITLE
`hydra clean-resources` for Azure for a user in dry-run mode doesn't show existing resources

### ISSUE THREAD

**Issue by dimakr** [scylladb/scylla-cluster-tests#8582]
Even though resources exist on Azure backend, the `hydra clean-resources --dry-run` command for a user doesn't list them.
For other backends (aws, gce, etc.) the resources are listed when `hydra clean-resources --dry-run` command is executed for the backend for a user.

Ex.:
- list resources for user
```
❯ hydra list-resources --user dmytro.kruglov
...
Checking Azure instances...
+------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                     Instances used on Azure                                                                      |
+-------------------------------------------------+-----------+----------------+--------------------------------------+----------------+---------------------------+
| Name                                            | Region-AZ | PublicIP       | TestId                               | RunByUser      | LaunchTime                |
+-------------------------------------------------+-----------+----------------+--------------------------------------+----------------+---------------------------+
| pr-provision-test-dmitriy-db-node-eastus-1      | eastus    | 52.149.152.138 | 3fae68fd-5b9a-4278-acc0-be3f72db7b79 | dmytro.kruglov | 2024-09-06 13:51:42+00:00 |
| pr-provision-test-dmitriy-db-node-eastus-2      | eastus    | 52.149.152.168 | 3fae68fd-5b9a-4278-acc0-be3f72db7b79 | dmytro.kruglov | 2024-09-06 13:51:45+00:00 |
| pr-provision-test-dmitriy-db-node-eastus-3      | eastus    | 52.149.152.171 | 3fae68fd-5b9a-4278-acc0-be3f72db7b79 | dmytro.kruglov | 2024-09-06 13:51:49+00:00 |
| pr-provision-test-dmitriy-loader-node-eastus-1  | eastus    | 52.255.140.93  | 3fae68fd-5b9a-4278-acc0-be3f72db7b79 | dmytro.kruglov | 2024-09-06 14:02:02+00:00 |
| pr-provision-test-dmitriy-monitor-node-eastus-1 | eastus    | 52.255.138.240 | 3fae68fd-5b9a-4278-acc0-be3f72db7b79 | dmytro.kruglov | 2024-09-06 14:04:24+00:00 |
+-------------------------------------------------+-----------+----------------+--------------------------------------+----------------+---------------------------+
```

- run clean-resources command for the user in dry-run mode
```
❯ hydra clean-resources --user dmytro.kruglov -b azure --dry-run
...
Cleaning everything for test id: 65e7ae16-5e9e-41cc-bcb6-9c22735c1829 in region: eastus
test id b0111597-eef5-4501-8863-b7a8e8c85783 from eastus - instances to clean: []
test id b1611080-4440-472a-9c5a-9e61b1728d80 from eastus - instances to clean: []
Cleaning everything for test id: 3fae68fd-5b9a-4278-acc0-be3f72db7b79 in region: eastus
test id eb5f5de5-0272-4dfe-8852-81ffef108e5b from eastus - instances to clean: []
Cleanup for the {'RunByUser': 'dmytro.kruglov', 'CreatedBy': 'SCT'} resources has been finished
```

The issue is rather cosmetic, but can sometimes confuse a user, when the list of resources available for cleanup in Azure is requested and it doesn't return any items.

**Comment by soyacz** [comment]
see
```
Cleaning everything for test id: 3fae68fd-5b9a-4278-acc0-be3f72db7b79 in region: eastus
```
In that case it's not showing individual resources.

**Comment by dimakr** [comment]
It differs from how it performs for other backends, e.g.:
```
hydra clean-resources --user dmytro.kruglov -b gce --dry-run
...
Done. Found total of 20 instances.
Going to delete: rolling-upgrade-artifacts--centos-9-db-node-46746c79-0-1 (gcp-sct-project-1 project)
Going to delete: rolling-upgrade-artifacts--centos-9-db-node-46746c79-0-2 (gcp-sct-project-1 project)
```

The same for aws.

So when a person would expect the same behavior for azure as well, it can confuse (it was with me, I understood that azure backend has my resources alive only when listed them directly, not when executed `clean-resources --dry-run`).

**Comment by soyacz** [comment]
dry-run for clean resources shows the action taken for cleaning resources. In terms of azure it would delete everything (meaning whole resource group) and dry-run shows exactly that - I don't see an issue with discrepancy between list-resources and clean-resources dry-run. 
In my opinion if someone wants to list resources should use list-resources, if wants to clean them, then clean-resources.
But I could be wrong, and should be easily fixable I suppose.

**Comment by fruch** [comment]
> dry-run for clean resources shows the action taken for cleaning resources. In terms of azure it would delete everything (meaning whole resource group) and dry-run shows exactly that - I don't see an issue with discrepancy between list-resources and clean-resources dry-run. In my opinion if someone wants to list resources should use list-resources, if wants to clean them, then clean-resources. But I could be wrong, and should be easily fixable I suppose.

Users are not completely aware for those differences between the backends, but output of dry-run is mostly cosmetic.

On the other hand for logging purpose in CI runs, having the resources listed might be helpful, and we iterate the VMs for reporting to Argus anyhow

**Comment by fruch** [comment]
> > > dry-run for clean resources shows the action taken for cleaning resources. In terms of azure it would delete everything (meaning whole resource group) and dry-run shows exactly that - I don't see an issue with discrepancy between list-resources and clean-resources dry-run. In my opinion if someone wants to list resources should use list-resources, if wants to clean them, then clean-resources. But I could be wrong, and should be easily fixable I suppose.
> > 
> > Users are not completely aware for those differences between the backends, but output of dry-run is mostly cosmetic.
> > 
> > On the other hand for logging purpose in CI runs, having the resources listed might be helpful, and we iterate the VMs for reporting to Argus anyhow
> 
> but do we iterate using 'clean-resources'? I doubt it.

So how does Argus get the report in Azure case ?

**Comment by soyacz** [comment]
`sdcm.utils.cloud_monitor.resources.instances.CloudInstances.get_azure_instances`
Definitely we have better methods than parsing dry-run output

**Comment by fruch** [comment]
> `sdcm.utils.cloud_monitor.resources.instances.CloudInstances.get_azure_instances`
> Definitely we have better methods than parsing dry-run output

Didn't suggest parsing the output 

I was asking how Argus gets the information that resources are clean, when it was as part of the clean resources stage ? 

If the answer is that it doesn't, then the data we have on Argus about nodes isn't very accurate, isn't it ?

**Comment by soyacz** [comment]
> > `sdcm.utils.cloud_monitor.resources.instances.CloudInstances.get_azure_instances`
> > Definitely we have better methods than parsing dry-run output
> 
> Didn't suggest parsing the output
> 
> I was asking how Argus gets the information that resources are clean, when it was as part of the clean resources stage ?
> 
> If the answer is that it doesn't, then the data we have on Argus about nodes isn't very accurate, isn't it ?

Yes, there's still some problem with marking Azure nodes upon clean-resources stage. I'll take a look at it.

**Comment by fruch** [comment]
@soyacz 

if the data in argus isn't getting updated on cleanups, we should take care of it.

if it's only the output of the command, it's a cosmetic issue from my POV

**Comment by soyacz** [comment]
reporting to argus does not work correctly - possibly there's an issue when whole RG is cleaned.

**Comment by fruch** [comment]
> reporting to argus does not work correctly - possibly there's an issue when whole RG is cleaned.

O.k. so I think we should attend to this one, since we do want reporting to be correct, it's causing user confusion now. (Happen several times in the last few months)
